apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -0,0 +1,117 @@
+import type { ControlProps, RankedTester } from "@jsonforms/core"
+import {
+  Box,
+  FormControl,
+  HStack,
+  Icon,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+} from "@chakra-ui/react"
+import { and, isStringControl, rankWith, schemaMatches } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import { FormLabel, Input, Tab } from "@opengovsg/design-system-react"
+import { BiEnvelopeOpen, BiFile, BiFileBlank, BiLink } from "react-icons/bi"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+
+export const jsonFormsLinkControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.LinkControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.format === "link"),
+  ),
+)
+
+export function JsonFormsLinkControl({
+  data,
+  label,
+  handleChange,
+  path,
+  description,
+  required,
+}: ControlProps) {
+  const LINK_TYPES = [
+    {
+      icon: BiFileBlank,
+      label: "Page",
+    },
+    {
+      icon: BiLink,
+      label: "External link",
+    },
+    {
+      icon: BiFile,
+      label: "File",
+    },
+    {
+      icon: BiEnvelopeOpen,
+      label: "Email",
+    },
+  ]
+  const dataString = data && typeof data === "string" ? data : ""
+
+  return (
+    <Box py={2}>
+      <FormControl isRequired={required}>
+        <FormLabel description={description}>{label}</FormLabel>
+
+        <Tabs variant="group" onChange={() => handleChange(path, "")}>
```

Comment:
Not really an issue, but does this mean that if I had data in a tab, and I clicked on another type of link(maybe to check if this suits the purpose better), and I clicked back my previous data will be lost?

Reply:
Yeah, the 4 types are mutually exclusive so the content of one type would not be transferrable to another, hence opted to clear the value to force the user to make the selection first.